File: cpp/007/007.cpp
#include <iostream>
#include <vector>
using namespace std;

bool is_prime (int num, vector<int>& primes)
{
	for (vector<int>::iterator i = primes.begin (); i != primes.end (); i++)
	  {
		if (num % *i == 0)
		  {
			return 0;
		  }
	  }

	return 1;
}

int main ()
{
	vector<int> primes;

	for (int i = 2; primes.size () != 10001; i++)
	  {
		if (is_prime (i, primes))
		  {
			primes.push_back (i);
		  }
	  }

	cout << primes.back () << endl;
	return 0;
}

File: jl/007/007.jl
#!/usr/bin/env julia

function is_prime(num, primes)
  !any(x-> (num % x == 0), primes)
end

function nth_prime(nth_val)
  primes = Set()
  num = 1
  while length(primes) < nth_val
    num += (num < 3) ? 1 : 2
    if is_prime(num, primes)
      union!(primes, num)
    end
  end
  num
end

println(nth_prime(10001))
